src/test/java/io/airlift/units/TestDataSize.java

```diff
@@ -120,7 +167,16 @@ public void testValueOf(String string, double expectedValue, DataSize.Unit expec
         DataSize size = DataSize.valueOf(string);
 
         assertEquals(size.getUnit(), expectedUnit);
-        assertEquals(size.getValue(), expectedValue);
+        assertThat(size.getValue()).isCloseTo(expectedValue, within(0.001));
+    }
+
+    @Test
+    public void testValueOfDecimalAndLongHandling()
+    {
+        //  If parsed as double, this results in only 2^53 not (2^53) + 1
+        DataSize tooLargeAsDouble = DataSize.ofBytes((1L << 53) + 1);
+        assertEquals(DataSize.valueOf(tooLargeAsDouble.toBytesValueString()), tooLargeAsDouble, "should parse as long and not double");
```

Comment:
Can we make this
```java
Long.toString(tooLargeAsDouble.toBytes()) + "B"
```
So that it's the same as the one below?

Reply:
Changed.